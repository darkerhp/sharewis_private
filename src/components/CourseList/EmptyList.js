import React from 'react';
import ReactNative from 'react-native';
import Hyperlink from 'react-native-hyperlink';
import I18n from 'react-native-i18n';

import BaseStyles from '../../baseStyles';
import redirectTo from '../../utils/linking';

const { StyleSheet, Text, View } = ReactNative;

const styles = StyleSheet.create({
  courseList: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  contentText: {
    ...BaseStyles.Text,
    textAlignVertical: 'center',
  },
  emptyList: {
    flex: 1,
    margin: 30,
  },
});


const emptyList = () =>
  <View style={styles.courseList}>
    <Hyperlink
      style={styles.emptyList}
      linkStyle={{ color: BaseStyles.hyperlink }}
      linkText={I18n.t('actWebsite')}
      onPress={redirectTo}
    >
      <Text style={styles.contentText}>
        {I18n.t('noCourses')}
      </Text>
    </Hyperlink>
  </View>;


export default emptyList;
