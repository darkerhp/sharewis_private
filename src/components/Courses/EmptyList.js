import React from 'react';
import ReactNative from 'react-native';
import Hyperlink from 'react-native-hyperlink';

import BaseStyles from '../../baseStyles';
import { ACT_API_URL } from '../../constants/Api';
import BaseTranslations from '../../translations';
import redirectTo from '../../utils/linking';

const { StyleSheet, Text, View } = ReactNative;

const styles = StyleSheet.create({
  courseList: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  contentText: {
    ...BaseStyles.Text,
    textAlignVertical: 'center',
  },
  emptyList: {
    flex: 1,
    margin: 30,
  },
});

const t = {
  ...BaseTranslations,
  noCourses: `購入済みのコースがない方は、 ${ACT_API_URL} よりお手続きを進めてください`,
};


const emptyList = () =>
  <View style={styles.courseList}>
    <Hyperlink
      style={styles.emptyList}
      linkStyle={{ color: BaseStyles.hyperlink }}
      linkText={t.actWebsite}
      onPress={redirectTo}
    >
      <Text style={styles.contentText}>
        {t.noCourses}
      </Text>
    </Hyperlink>
  </View>;


export default emptyList;
