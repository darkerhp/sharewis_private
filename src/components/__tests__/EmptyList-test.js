/* eslint-disable no-undef */
/* eslint-disable import/no-extraneous-dependencies */
import 'react-native';
import renderer from 'react-test-renderer';
import React from 'react';
import EmptyList from '../CourseList/EmptyList';

jest.mock('react-native-loading-spinner-overlay', () => 'Spinner');
jest.mock('../../utils/linking', () =>
  () => 'redirectTo',
);


describe('EmptyList', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <EmptyList isFetching />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
