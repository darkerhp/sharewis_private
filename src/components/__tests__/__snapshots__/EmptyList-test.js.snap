// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyList renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    linkStyle={false}
    linkText="[missing \\"en.actWebsite\\" translation]"
    onPress={[Function]}
    style={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#222",
          "fontSize": 17,
          "lineHeight": 22,
          "margin": 30,
          "textAlign": "center",
        }
      }
    />
  </View>
</View>
`;
